### DISCUSSION TITLE
Hub and spoke carrot-mapper

### DISCUSSION THREAD

**Discussion by prquinlan** [Health-Informatics-UoN/carrot#59] [Ideas]
Can we have a carrot mapper that is installed within an organisation infra, but can still benefit from mapping rules from the central system

**Comment by spco** [comment]
Perhaps architected as multiple Mapper instances, all of which can feed mapping rules into/ingest mapping rules from a central repo.

Rules format (a trimmed-down version of the existing JSON?) would need clarifying, as well as permissions on the central system, and whether it's an add-on to a central Mapper instance or a stand-alone tool.

**Comment by AndyRae** [comment]
I've moved this to a discussion - as it needs wider thought, design, and planning. 
I have some initial ideas around this which I will share as well.

**Comment by AndyRae** [comment]
This feels especially important to support governance, and distributed compute.

The absolute minimum version of it is to enable importing `json` files into Carrot as rules.
The next level is importing from a repository - like Github. 

Both of these might be enough - but will probably end up with manual rules management hell.

The level above that is having a central repository (a standalone API), where Carrot instances can push/pull rules from.

Several things would help with this:
- Health-Informatics-UoN/carrot#57  - (SSSOM has a standard for a group of rules already, and [repository API](https://github.com/mapping-commons/sssom-api) that might be no longer maintained) 
- Health-Informatics-UoN/carrot-mapper#845

A central repository would not be much work itself (a database, with permissions, a limited set of endpoints). An integration with Carrot would be more work, but eased if the above were in place.
A central repository could also enable: Health-Informatics-UoN/carrot-mapper#847 

We would make the assumption that there can be many Carrots attaching to many hubs.
